Assets/Scripts/Managers/BGMManager.cs

```diff
@@ -46,6 +46,19 @@ private void OnDestroy()
             SceneManager.sceneLoaded -= OnSceneLoaded;
         }
 
+        /// <summary>
+        /// Play the given background music.
+        /// </summary>
+        /// <param name="audioClip">Audio clip for the background music.</param>
+        public void PlayBGM(AudioClip audioClip)
```

Comment:
nit: if we're creating a new method here might as well use it (e.g. line 94-98, 32-35)